src/libraries/Common/src/System/Net/SocketProtocolSupportPal.Unix.cs

```diff
@@ -6,59 +6,21 @@
 using System.Net.Internals;
 using System.Net.Sockets;
 using System.Runtime.InteropServices;
-using System.Threading;
 
 namespace System.Net
 {
-    internal class SocketProtocolSupportPal
+    internal static class SocketProtocolSupportPal
     {
-        private static bool s_ipv4 = true;
-        private static bool s_ipv6 = true;
-
-        private static bool s_initialized;
-        private static readonly object s_initializedLock = new object();
-
-        public static bool OSSupportsIPv6
-        {
-            get
-            {
-                EnsureInitialized();
-                return s_ipv6;
-            }
-        }
-
-        public static bool OSSupportsIPv4
-        {
-            get
-            {
-                EnsureInitialized();
-                return s_ipv4;
-            }
-        }
-
-        private static void EnsureInitialized()
-        {
-            if (!Volatile.Read(ref s_initialized))
-            {
-                lock (s_initializedLock)
-                {
-                    if (!s_initialized)
-                    {
-                        s_ipv4 = IsProtocolSupported(AddressFamily.InterNetwork);
-                        s_ipv6 = IsProtocolSupported(AddressFamily.InterNetworkV6);
-
-                        Volatile.Write(ref s_initialized, true);
-                    }
-                }
-            }
-        }
+        public static bool OSSupportsIPv6 { get; } = IsProtocolSupported(AddressFamily.InterNetworkV6);
+        public static bool OSSupportsIPv4 { get; } = IsProtocolSupported(AddressFamily.InterNetwork);
+        public static bool OSSupportsUnixDomainSockets { get; } = IsProtocolSupported(AddressFamily.Unix);
 
         private static unsafe bool IsProtocolSupported(AddressFamily af)
         {
             IntPtr socket = (IntPtr)(-1);
             try
             {
-                Interop.Error err = Interop.Sys.Socket(af, SocketType.Dgram, (ProtocolType)0, &socket);
+                Interop.Error err = Interop.Sys.Socket(af, SocketType.Dgram, 0, &socket);
```

Comment:
Not familiar with Winsock, but the Winsock version has been changed to use SOCK_STREAM instead of SOCK_DGRAM.  Is that on purpose?

Reply:
Yes, dgram doesn't work with UDS here.